src/server/pegasus_write_service_impl.h

```diff
@@ -806,6 +833,23 @@ class pegasus_write_service::impl : public dsn::replication::replica_base
         return expire_ts;
     }
 
+    dsn::error_code get_external_files_path(const std::string &bulk_load_dir,
+                                            const dsn::replication::bulk_load_metadata &metadata,
+                                            /*out*/ std::vector<std::string> &files_path)
+    {
+        for (const auto &f_meta : metadata.files) {
+            // const dsn::replication::file_meta &f_meta = *file;
+            const std::string &file_name =
+                dsn::utils::filesystem::path_combine(bulk_load_dir, f_meta.name);
+            if (dsn::utils::filesystem::verify_file(file_name, f_meta.md5, f_meta.size)) {
```

Comment:
Why should you verfiy these files again?

Reply:
It's for double check, it will cost long time during files downloaded and ingestion.